src/test/java/codeu/model/store/basic/UserStoreTest.java

```diff
@@ -0,0 +1,92 @@
+package codeu.model.store.basic;
+
+import codeu.model.data.User;
+import codeu.model.store.persistence.PersistentStorageAgent;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+public class UserStoreTest {
+
+  private UserStore userStore;
+  private PersistentStorageAgent mockPersistentStorageAgent;
+
+  private final User USER_ONE =
+      new User(UUID.randomUUID(), "test_username_one", Instant.ofEpochMilli(1000));
+  private final User USER_TWO =
+      new User(UUID.randomUUID(), "test_username_two", Instant.ofEpochMilli(2000));
+  private final User USER_THREE =
+      new User(UUID.randomUUID(), "test_username_three", Instant.ofEpochMilli(3000));
+
+  @Before
+  public void setup() {
+    mockPersistentStorageAgent = Mockito.mock(PersistentStorageAgent.class);
+    userStore = UserStore.getTestInstance(mockPersistentStorageAgent);
+
+    List<User> userList = new ArrayList<>();
+    userList.add(USER_ONE);
+    userList.add(USER_TWO);
+    userList.add(USER_THREE);
+    userStore.setUsers(userList);
+  }
+
+  @Test
+  public void testGetUser_byUsername_found() {
+    User resultUser = userStore.getUser(USER_ONE.getName());
+
+    assertEquals(USER_ONE, resultUser);
+  }
+
+  @Test
+  public void testGetUser_byId_found() {
+    User resultUser = userStore.getUser(USER_ONE.getId());
+
+    assertEquals(USER_ONE, resultUser);
+  }
+
+  @Test
+  public void testGetUser_byUsername_notFound() {
+    User resultUser = userStore.getUser("fake username");
+
+    Assert.assertNull(resultUser);
+  }
+
+  @Test
+  public void testGetUser_byId_notFound() {
+    User resultUser = userStore.getUser(UUID.randomUUID());
```

Comment:
Technically, this UUID could collide with one previously generated, causing a spurious test failure. This could be fixed by preallocating a UUID, and making sure it does not match any of the User UUIDs created by setup().

Reply:
I thought we were assuming that UUID always generated a unique ID? If not we need to rethink more than just this test. :fearful:

Reply:
Indeed. My understanding is that the Java UUID is based on a random number generator, so a collision is possible, but so unlikely that in practice it can be ignored. Still it's a good idea to think about the possible impact of an RNG collision (small, in our case), and consider whether it is worthwhile adding additional safety checks (probably not).

Reply:
Yeah, agreed. I think it's worth thinking harder about, but probably not for this specific test.